openedx/core/djangolib/model_mixins.py

```diff
@@ -12,3 +12,23 @@ def __init__(self, *args, **kwargs):  # pylint: disable=unused-argument
         Override to kill usage of this model.
         """
         raise TypeError("This model has been deprecated and should not be used.")
+
+
+class RetireUserByEmailMixin:
+    """
+    Mixin allows inheriting models to delete users by their email field. A 
+    Jenkins job that removes user data calls a model's "retire_user".
```

Comment:
...and I don't think we need to worry about describing the Jenkins aspect of things in this docstring.